tools/kbuild/Makefile.kbuild

```diff
@@ -557,6 +561,10 @@ firmware-objs	:= $(patsubst %,%/built-in.o, $(objs-y))
 firmware-libs	:= $(patsubst %,%/lib.a, $(libs-y))
 firmware-all	:= $(firmware-objs) $(firmware-libs)
 
+ifneq ($(OOT_USES_CXX),)
+	firmware-libs += -lstdc++
```

Comment:
Note to the reviewer: we need this because we also pass `-nostdlib` to the linker for plain C-based apps, so it won't link to `stdc++` by default. However, we apparently need `stdc++` for the `new` operator to work.

I also tried adding this to `LDFLAGS` instead, but then it comes too early in the command line that eventually calls the linker. It had to be placed just before where we have `-lm` there (see a few lines below).